lightly/active_learning/agents/agent.py

```diff
@@ -190,14 +190,15 @@ def query(self,
         # calculate active learning scores
         scores_dict = None
         if al_scorer is not None:
+            scores_dict = al_scorer.calculate_scores()
             no_query_samples = len(self.query_set)
-            no_query_samples_with_scores = len(al_scorer.model_output)
+            no_query_samples_with_scores = len(list(scores_dict.values())[0])
```

Comment:
You could make this check more complete with something like this:

```python
if any(len(list(scores)) != no_query_samples for scores in scores_dict.values):
    raise...

```

Reply:
Yes, this helps in case our Scorers work incorrectly.